veps/sig-storage/dic-on-heterogeneous-cluster/dic-on-heterogeneous-cluster.md

```diff
@@ -0,0 +1,538 @@
+be SSP to create multiple `DataImportCron` CRs for each multi-arch `DataImportCronTamplate`.
+
+Below are too additional suggestions for the same question.
+
+#### Create Multiple `DataImportCronTemplate` Objects, One for Each Architecture
+HCO could create a new `DataImportCronTemplate` object for each architecture in the `SSP` CR, and the user will be
+responsible for doing the same for custom `DataImportCronTemplate` objects.
+
+pros:
+
+- no changes are required from SSP, except for adopting the new CDI API, and changes regarding the VM templates.
+
+cons:
+
+- the user will be responsible for creating and maintaining the `DataImportCronTemplate` objects, and
+  the user will need to create a new `DataImportCronTemplate` object for each architecture.
+- the `SSP` CR will be huge, and will contain a lot of `DataImportCronTemplate` objects. It will be hard to maintain it
+  or understand it.
+- it looses the meaning of the templating mechanism, by the `DataImportCronTemplate` type.
+
+#### Create Multiple `DataSource` CRs from a Single `DataImportCron` Object
+No changes in SSP, regarding the `DataImportCronTemplate` objects. SSP will create a single `DataImportCron` CR for each
+`DataImportCronTemplate` object in the `SSP` CR, as done today. The `ssp.kubevirt.io/dict.architectures` (with a new
+name, e.g. `cdi.kubevirt.io/dict.architectures`) annotation will be copied to the `DataImportCron` CR, as any other
+annotation.
+
+CDI will create multiple `DataSource` CRs, one for each architecture, according to the
+`cdi.kubevirt.io/dict.architectures` annotation, and will import the images into architecture specific `VolumeSnapshot`
+resources. 
+
+When performing the actual image import, if the `cdi.kubevirt.io/dict.architectures` annotation is set in
+the `DataImportCron` CR, the CDI will create architecture specific `DataVolume` CR, for each architecture
+listed in the annotation. CDI will set the `DataVolome` CR's `spec.source.registry.platform.architecture` field, to the
+required architecture.
+
+CDI will add the `template.kubevirt.io/architecture` label to the `DataSource` CR, with the architecture of the image,
+and will set the `cdi.kubevirt.io/storage.import.datasource-name` label to the value of the `spec.managedDataSource` 
+field in the `DataImportCron` CR.
+
+CDI will need to remove any orphan `DataSource` or `VolumeSnapshot` CRs, that are not referenced by any `DataImportCron`
+CR.
+
+SSP will still need to create architecture specific templates, and will still need to set the `DataSource` name 
+in each template, to the architecture specific `DataSource` CR name.
+
+Open questions:
+1. SSP currently creates the `DataSource` CRs in some cases. We have two options here:
+   1. SSP will have to have the same logic as in CDI, when creating the architecture specific `DataSource` CRs.
+   2. SSP will no longer create the `DataSource` CRs, and will let CDI to do that.
```

Comment:
SSP creates DataSources for all templates it deploys. If we set the default name of the DataSources in the templates accordingly to the logic that CDI will be using for associating a DataImportCron with a DataSource we should be fine.

Reply:
My concern is that eventually either SSP or CDI will change the implementation a bit, and they won't match anymore.